src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -346,6 +349,8 @@ public static CloseableHttpResponse execute(
         }
 
         retryCount++;
+        lastStatusCodeForRetry =
```

Comment:
When there is no response or status code, the value of `retry` will be the string value "null". However, we should never hit this case because the JDBC driver does not retry if the response is null.

Reply:
@sfc-gh-mknister have you tested the case where there is no response from the backend resulting in a timeout? What is the error code in that case? Also instead of using null can we choose a number that is less risky? Like 0? Is 0 already used for something?

Reply:
@sfc-gh-hchaturvedi if there is no response from the backend, the response will come back as null. I tested this manually by using Burpsuite and intercepting the POST requests, and then waiting until the driver timed out. In this case, it is possible to do a retry.

Looking at the list of [status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status), it looks like 0 is not used for anything! Changed the code to `0` for the null case instead.